src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -476,7 +492,9 @@ private static boolean isBlockAloneOnSingleLine(Details details) {
      * @return true if the right curly is followed by a semicolon.
      */
     private static boolean isRightcurlyFollowedBySemicolon(Details details) {
-        return details.nextToken.getType() == TokenTypes.SEMI;
```

Comment:
just a null check will be better here instead of using optionals

Reply:
Null check would be out of coverage. Would you like me to add this check + coverage for this condition.

Reply:
We can just remove null check since nextToken would atmost be `EOF` and `Parser` would through an exception in that case much before.